javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/AbstractJavaParserContext.java

```diff
@@ -166,7 +166,7 @@ public final Context getParent() {
             try {
                 typeOfScope = JavaParserFacade.get(typeSolver).getType(scope);
             } catch (Exception e) {
-                throw new RuntimeException("Issue calculating the type of the scope of " + this, e);
+                throw new UnsolvedSymbolException("Issue calculating the type of the scope of " + this);
```

Comment:
I agree it could help, let's see what @ftomassetti has to say about it. :-)

Reply:
I would say this depends on the type of the exception. What about having a specific catch for meaningful resolution exceptions and a generic one that throws a RuntimeException?

Reply:
I would argue that throwing a generic `RuntimeException` is *never* a good idea. Ideally, every thrown exception should be meaningful. The code needs to figure out what went wrong and throw a sensible exception accordingly. Throwing a generic `RuntimeException` is a bit like a segfault or something... the application just crashes and does not give any specific information to the user.

In addition, generic catches (`catch(Exception e)`) are generally not a good idea in my opinion, because it completely obscures what kind of exceptions the programmer expects to be thrown. In case several different types of exceptions are expected to be potentially thrown that need to be handled in the same way, one should use a multi catch (`catch(XException | YException | ZException e)`).

So in this case, I am not particularly fond of the `catch(Exception e)` (but that was already there before and I do not know what types of exceptions you expect may happen), but I think it is fair to handle all exceptions at this point by rethrowing an `UnsolvedSymbolException`, since that is effectively what it means: the type of the scope could not be solved for some reason. :-)

Reply:
My understanding is that we should handle only exceptions to which we can react meaningfully in that context, so I would agree we could capture some exceptions and throw UnsolvedSymbolException in reaction.

As you said throwing a RuntimeException is suboptimal but the reason for doing that is to decorate it with additional information. It helped debugging because the code is complex and when there was an exception I could follow what was happening by looking at the information added into the exception ("oh, I got this issue while I was trying to solve this symbol in this context and that was caused by this other problem..."). So for me throwing a RuntimeException was almost as I did not catch it: I just saw it passing, added some debugging infor and resend it on its way.

Reply:
Why not going back to what @ThLeu suggested and add a constructor to UnsolvedSymbolException which preserve the root cause (i.e., the other exception)? So you get the UnsolvedSymbolException and I get a way to debug